tests/rptest/tests/delete_records_test.py

```diff
@@ -772,10 +772,10 @@ class PartitionMoveExceptionReporter:
         partition_move_thread.join()
         self.redpanda.logger.info("Joining on producer thread")
         producer.wait()
-        self.redpanda.logger.info("Calling consumer::stop")
-        consumer.stop()
         self.redpanda.logger.info("Joining on consumer thread")
         consumer.wait()
+        self.redpanda.logger.info("Calling consumer::stop")
```

Comment:
This seems to fix https://redpandadata.atlassian.net/browse/CORE-7770, thank you!